eis/run_models.py

```diff
@@ -413,6 +413,75 @@ def evaluations(self, predictions_proba, predictions_binary, test_y, model_id, t
         db_conn.close()
         return None
 
+    def individual_feature_ranking(self, fitted_model, test_matrix, model_id, test_date):
+        ###################
+        # This method is a beta version tested for top k optimized random forests
+        ###################
+
+        # get the list of all features in the test matrix
+        feature_list = test_matrix.columns
+
+        # extract all global feature importance values from the tree
+        importance_dict = {}
+        for j, value in enumerate(fitted_model.feature_importances_):
+            if value > 0:
+                importance_dict[feature_list[j]] = value
+
+        # filter out dummies
+        importance_dict_filtered = {k: v for k, v in importance_dict.items() if not 'dummy' in k}
+
+        rftree_feature_list = sorted(importance_dict_filtered, key=importance_dict_filtered.get, reverse=True)
+
+        # create a new text matrix with float conversion (panda stores some columns from postgres as decimal object)
+        tmp_test = test_matrix.applymap(lambda x: float(x))
+        test_matrix_reduced = pd.DataFrame()
+
+        # add the top 30 features from the RandomForest used without dummies
+        for idx, feature in enumerate(rftree_feature_list):
+            if idx >= 30:
```

Comment:
A more pythonic way of doing this will be:

```
for feature in rftree_feature_list[:30]:
    test_matrix_reduced[feature] = tmp_test[feature]
```

Also, it is better if `30` is an argument in the method...